lib/ansible/modules/cloud/docker/docker_swarm.py

```diff
@@ -427,11 +436,26 @@ def inspect_swarm(self):
             data = self.client.inspect_swarm()
             json_str = json.dumps(data, ensure_ascii=False)
             self.swarm_info = json.loads(json_str)
+
             self.results['changed'] = False
             self.results['swarm_facts'] = self.swarm_info
+
+            unlock_key = self.get_unlock_key()
+            self.swarm_info.update(unlock_key)
         except APIError:
             return
 
+    def get_unlock_key(self):
+        default = {'UnlockKey': None}
+        if not self.has_swarm_lock_changed():
+            return default
+        return self.client.get_unlock_key() or default
```

Comment:
Does `self.client.get_unlock_key()` returns something of form `{ 'UnlockKey': xxx }` (as this function seems to assume), or does it just return `xxx` (as the function in `docker_swarm_info` seems to assume)?

Reply:
It does return `{ 'UnlockKey': xxx }`. The method in `docker_swarm_info` actually unpacks that dictionary:

```python
def get_docker_swarm_unlock_key(self):
    unlock_key = self.client.get_unlock_key()
    return unlock_key.get('UnlockKey')
```

In my opinion it was a cleaner API on docker_swarm_info to just return the key in the variable `swarm_unlock_key` while it made more sense in `docker_swarm` to return `UnlockKey` along with the other tokens (`JoinTokens`) in the same dictionary .